osquery/tables/system/hash.cpp

```diff
@@ -140,7 +167,96 @@ std::string hashFromFile(HashType hash_type, const std::string& path) {
   }
 }
 
-namespace tables {
+/*
+ * FileHashCache implements persistent in-memory caching for files' hashes.
+ * this cache has LRU eviction policy. the hash is recalculated
+ * every time the mtime or size of the file changes.
+ */
+
+#if defined(WIN32)
+
+#define stat _stat
+
+#endif
+
+struct FileHashCache {
+  time_t file_mtime;
+  off_t file_size;
+  time_t cache_access_time;
+  MultiHashes hashes;
+  // tracking it for eviction policy
```

Comment:
nit, for doxygen docs generation `make docs`, use `///` for single-line documentation and the long-form docblock: https://github.com/facebook/osquery/blob/master/include/osquery/database.h#L105 for multi-line (recommended for methods).